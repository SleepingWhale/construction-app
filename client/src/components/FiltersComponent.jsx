import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';


export class FiltersComponent extends Component {
  static propTypes = {
    filters: PropTypes.arrayOf(PropTypes.string).isRequired,
    onChange: PropTypes.func.isRequired,
  };

  checked = {};

  onInputChange = (e) => {
    const { name, checked } = e.target;
    const { onChange } = this.props;
    const result = [];
    this.checked[name] = checked;

    Object.entries(this.checked).forEach(([name, isChecked]) => {
      if (isChecked) result.push(name);
    });

    onChange(result);
  };

  render() {
    const { filters } = this.props;

    return (
      <Fragment>
        {
          filters.map((f) => (
            <div className="form-check form-check-inline" key={f}>
              <span className="badge badge-info p-2">
                <input type="checkbox" className="form-check-input" id={f} name={f} onChange={this.onInputChange}/>
                <label className="form-check-label" htmlFor={f}>{f}</label>
              </span>
            </div>
          ))
        }
      </Fragment>
    )
  }
}
