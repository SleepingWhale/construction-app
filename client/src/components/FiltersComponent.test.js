import React from 'react';
import { shallow } from 'enzyme';
import { FiltersComponent } from './FiltersComponent';


const filters = ['a', 'b', 'c'];
const onChangeHandler = jest.fn();



describe('FiltersComponent', () => {
  const component = shallow(<FiltersComponent filters={filters} onChange={onChangeHandler}/>);

  it('should render correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('should call onChangeHandler with proper arguments', () => {
    const aCheck = component.find('#a');
    const bCheck = component.find('#b');

    aCheck.simulate('change', { target: { checked: true, name: 'a' } });
    bCheck.simulate('change', { target: { checked: true, name: 'b' } });
    aCheck.simulate('change', { target: { checked: false, name: 'a' } });

    expect(onChangeHandler).toHaveBeenNthCalledWith(1, ['a']);
    expect(onChangeHandler).toHaveBeenNthCalledWith(2,['a', 'b']);
    expect(onChangeHandler).toHaveBeenNthCalledWith(3,['b']);
  });
});
