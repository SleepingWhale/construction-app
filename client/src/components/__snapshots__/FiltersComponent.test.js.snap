// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FiltersComponent should render correctly 1`] = `
<Fragment>
  <div
    className="form-check form-check-inline"
    key="a"
  >
    <span
      className="badge badge-info p-2"
    >
      <input
        className="form-check-input"
        id="a"
        name="a"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="form-check-label"
        htmlFor="a"
      >
        a
      </label>
    </span>
  </div>
  <div
    className="form-check form-check-inline"
    key="b"
  >
    <span
      className="badge badge-info p-2"
    >
      <input
        className="form-check-input"
        id="b"
        name="b"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="form-check-label"
        htmlFor="b"
      >
        b
      </label>
    </span>
  </div>
  <div
    className="form-check form-check-inline"
    key="c"
  >
    <span
      className="badge badge-info p-2"
    >
      <input
        className="form-check-input"
        id="c"
        name="c"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="form-check-label"
        htmlFor="c"
      >
        c
      </label>
    </span>
  </div>
</Fragment>
`;
